addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.survey.controllers.main import Survey
+from odoo.addons.survey.tests import common
+from odoo.addons.website.tools import MockRequest
```

Comment:
Hmm, ok, this is needed even though survey does not depend on website.

Can you double-check that this test passes without website installed?
If so we can keep it, even if not ideal.

Reply:
If I don't use the MockRequest I get a `Runtime: object unbound` exception because it cannot find the `request` object.
If I drop db et only install with `-i survey` it works. :slightly_smiling_face: 

When searching for something similar, in the `TestSignControllerCommon` class in the `Sign` module (enterprise), they also are using the MockRequest to test a Controller method and it doesn't have website or web in the manifest : 
```python
    'depends': ['mail', 'attachment_indexation', 'portal', 'sms'],
```
But I don't know if it's something that we can do or not since it's the only occurence that I've found.

===============================================================

In the `web_studio` module, they are using this kind of thing in 3 different test files : 
```python
    def setUp(self):
        super(TestReportEditor, self).setUp()
        _request_stack.push(self)  # crappy hack to use a fake Request
        self.WebStudioController = WebStudioController()
        
     def tearDown(self):
        super(TestReportEditor, self).tearDown()
        _request_stack.pop()
 ```
 It's working, all the tests passed with correct values and there is only the self on the `request_stack` on tearDown so `pop` is correctly removing `self`.
 It looks like a good hack ? :thinking: I choose this one in my commit

Reply:
Well actually I would prefer using MockRequest as it seems a lot more complete / robust.
(And it does not include a "crappy hack" :eyes: ).

It's not ideal in terms of modularity, but as you said it's already done in sign.